components/wmtk_components/delaunay/internal/DelaunayOptions.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <spdlog/spdlog.h>
+#include <nlohmann/json.hpp>
+
+namespace wmtk {
+namespace components {
+namespace internal {
+
+struct DelaunayOptions
+{
+    std::string type;
+    std::string input;
+    std::string output;
+    long cell_dimension = -1;
+};
+
+NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(DelaunayOptions, type, input, output, cell_dimension);
```

Comment:
non-intrusive implies it's not adding it to the struct. usually these things are implemented like

```
template <typename T>
json to_json(const T& t) {
   return t.to_json();
}

#define NONINTRUSIVE_IMPL(T, input)  template <> json to_json<T>(const T& t) { json; json[#input] = t.input; }
```
if the nonintrusive is only created in a single compilation (or used with an anonymous namespace) or whatnot it can be put in a cpp?